### PR TITLE
Split WalletConnect v1 and v2 reducers

### PR DESCRIPTION
**Pull request by jeanregisser**
### Description

Split WalletConnect v1 and v2 reducers so it's easier to maintain.
Also fixes bugs when interacting with both v1 and v2.
And some regressions around wallet address and analytics due to a bad merge.

### CODE REVIEW

**Review by gnardini — COMMENTED** [review]
This looks great, it will be so much easier to work with this with these changes, thanks so much 😄

File: packages/mobile/src/walletConnect/request.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ export function* handleRequest({ method, params }: { method: string; params: any
       const normalizer = new TxParamsNormalizer(kit.connection)
       const tx: CeloTx = yield call(normalizer.populate.bind(normalizer), params[0])
       const sendTxMethod: SendTransactionMethod = function* () {
+        // TODO: this was needed to fix PoolTogether, we shouldn't use it for all requests
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply packages/mobile/src/walletConnect/request.ts:47]: Yes the gas estimation was needed to make PoolTogether work.
I'll update the comment.

File: packages/mobile/src/walletConnect/v1/saga.ts [hunk 2]
```diff
@@ -280,3 +335,11 @@ export function* walletConnectV1Saga() {
 
   yield call(checkPersistedState)
 }
+
+export function* initialiseWalletConnectV1(uri: string, origin: WalletConnectPairingOrigin) {
+  // if (!client) {
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply packages/mobile/src/walletConnect/v1/saga.ts:340]: This needs to be added, I left it as a reminder to have something similar to what we had in v2.

File: packages/mobile/src/walletConnect/v2/saga.ts [hunk 3]
```diff
@@ -343,7 +441,7 @@ export function* handleInitialisePairing({ uri, origin }: InitialisePairing) {
       origin,
     })
     if (!client) {
-      throw new Error('missing client')
+      throw new Error(`missing client`)
```
[the remark this answers is not in the document]
Comment (jeanregisser) [reply packages/mobile/src/walletConnect/v2/saga.ts:444]: Yep, would be nice to use eslin't to make this automatic.

**Pull request merged**